more_itertools/more.py

```diff
@@ -181,7 +188,13 @@ def __getitem__(self, index):
         if isinstance(index, slice):
             return self._get_slice(index)
 
-        return self._get_slice(slice(index, index + 1, None))[0]
+        cache_len = len(self._cache)
+        if index < 0:
```

Comment:
Looks good to me. I twiddled around with it but couldn't find any off-by-ones. I *feel* like this if/elif wants to be made one with the similar one in _get_slice(), but it's not obvious that it would result in a simplification. If you don't immediately say "Of course! Like this!", then let's declare victory and merge as-is. Nice work, yet again!